### PR TITLE
Dc apps

### PR DESCRIPTION
**Pull request by thast**
@micmitch Apps:
- Plate 2D, 2.5D
- Resistive layer
- Conversion to dipole

@thast:
-Cylinder 2D

### CODE REVIEW

File: em_examples/DCWidgetPlate_2D.py [hunk 1]
```diff
@@ -298,67 +312,135 @@ def sumPlateCharges(xc,zc,dx,dz,rotAng,qSecondary):
     return qPosSum, qNegSum, qPosAvgLoc, qNegAvgLoc
 
 
-# Inline functions for computing apparent resistivity
-eps = 1e-9 #to stabilize division
-G = lambda A, B, M, N: 1. / ( 1./(np.abs(A-M)+eps) - 1./(np.abs(M-B)+eps) - 1./(np.abs(N-A)+eps) + 1./(np.abs(N-B)+eps) )
-rho_a = lambda VM,VN, A,B,M,N: (VM-VN)*2.*np.pi*G(A,B,M,N)
+def getSensitivity(survey,A,B,M,N,model):
```
Comment (lheagy) [inline em_examples/DCWidgetPlate_2D.py:229]: @thast: do you have a linter on for python? https://github.com/SublimeLinter/SublimeLinter-pep8

In the future, if you would use that, that would be helpful. It does not need to be implemented for this pr though

**Comment by lheagy** [comment]
@thast, @sgkang, @micmitch, can you give a 👍 if you are good with merging this in?

**Comment by thast** [comment]
@lheagy, @sgkang  : I think I have found a solution for your scale problems! I will update DC_cylinder_2D with it tonight and if it works for you I can make it all the DCapps!
(Log secondary sensitivity is quite fun to look at)

**Comment by lheagy** [comment]
:+1: can you share a picture??

**Comment by sgkang** [comment]
Nice work @thast @micmitch !! I have few minor suggestions, but other than that I am happy with this pull request. 

For `DCResLayer2D` and `DCResLayer2_5D`

- [x]  In set up figures notation for resistivities are: rho_1, rho_2, rho_3, but in the app they are not matching (rho_half, rho_target, rho_cyl)

- [x]  When "Sensitivity" is chosen, Pole-Dipole and Pole-Pole are not working (we can fix this later)

**Comment by micmitch** [comment]
Just made those changes for you Seogi.

We are lacking sensitivity for the 2.5D apps since there is no fields object in the 2.5D class. This fields object is needed in the call to JTvec. We could display the 2D sensitivity in the 2.5D app but this might be misleading.

**Comment by thast** [comment]
😃

**Pull request merged**